msgpack.sh

```diff
@@ -0,0 +1,40 @@
+package: msgpack
+version: cpp-2.1.5
+source: https://github.com/msgpack/msgpack-c
+build_requires:
+ - CMake
+ - "GCC-Toolchain:(?!osx)"
+prefer_system: "(?!slc5)"
+prefer_system_check: |
+  printf "#include <msgpack/version.hpp>\n#define VERSION (MSGPACK_VERSION_MAJOR * 10000) + (MSGPACK_VERSION_MINOR * 100) + MSGPACK_VERSION_REVISION\n#if(VERSION < 20105)\n#error \"msgpack version >= 2.1.5 needed\"\n#endif\nint main(){}" | c++ -I$(brew --prefix msgpack)/include -xc++ -std=c++11 - -o /dev/null
+---
+mkdir -p $INSTALLROOT
+
+cmake $SOURCEDIR                                                 \
+      ${C_COMPILER:+-DCMAKE_C_COMPILER=$C_COMPILER}              \
+      ${CXX_COMPILER:+-DCMAKE_CXX_COMPILER=$CXX_COMPILER}        \
+      -DMSGPACK_CXX11=ON                                         \
+      -DMSGPACK_BUILD_TESTS=OFF                                  \
+      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT
+
+cmake --build . --target install ${JOBS:+-- -j$JOBS}
+
+#ModuleFile
+mkdir -p etc/modulefiles
+cat > etc/modulefiles/$PKGNAME <<EoF
+#%Module1.0
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0
+# Our environment
+setenv MSGPACK_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
+prepend-path LD_LIBRARY_PATH \$::env(MSGPACK_ROOT)/lib
```

Comment:
No executables in `bin`? Can you confirm?

Reply:
What we can do is, by default, building it with `-DBUILD_NANOMSG_TRANSPORT=OFF` with a switch that enables it if and only if `msgpack` and `nanomsg` are dependencies. I don't recall the reason.

Reply:
Ok, since I do not know at the moment how to implement such a switch + condition, I'll leave the dependency list as it is for this PR. When we do the change on FairRoot recipe level, someone on O2 side should anyways decide, which FairMQ components are built by default. Another switch to consider is `-DBUILD_DDS_PLUGIN=ON/OFF`.